lntest/itest/lnd_routing_test.go

```diff
@@ -1066,6 +1066,137 @@ func testPrivateChannels(net *lntest.NetworkHarness, t *harnessTest) {
+
+	// Open a channel with 100k satoshis between Alice and Bob.
+	ctxt, _ := context.WithTimeout(ctxb, channelOpenTimeout)
+	chanPointAliceBob := openChannelAndAssert(
+		ctxt, t, net, net.Alice, net.Bob,
+		lntest.OpenChannelParams{
+			Amt: chanAmt,
+		},
+	)
+	defer closeChannelAndAssert(
+		ctxt, t, net, net.Alice, chanPointAliceBob, false,
+	)
+
+	// Get Alice's funding point.
+	aliceChanTXID, err := lnrpc.GetChanPointFundingTxid(chanPointAliceBob)
+	require.NoError(t.t, err, "unable to get txid")
+	aliceFundPoint := wire.OutPoint{
+		Hash:  *aliceChanTXID,
+		Index: chanPointAliceBob.OutputIndex,
+	}
+
+	// Create a new node Carol.
+	carol, err := net.NewNode("Carol", nil)
+	require.NoError(t.t, err, "unable to create node carol")
+	defer shutdownAndAssert(net, t, carol)
+
+	// Connect Carol to Bob.
+	ctxt, _ = context.WithTimeout(ctxb, defaultTimeout)
+	require.NoError(t.t,
+		net.ConnectNodes(ctxt, carol, net.Bob),
+		"unable to connect carol to bob",
+	)
+
+	// Open a channel with 100k satoshis between Bob and Carol.
+	ctxt, _ = context.WithTimeout(ctxb, channelOpenTimeout)
+	chanPointBobCarol := openChannelAndAssert(
+		ctxt, t, net, net.Bob, carol,
+		lntest.OpenChannelParams{
+			Amt:     chanAmt,
+			Private: true,
+		},
+	)
+	defer closeChannelAndAssert(
+		ctxt, t, net, net.Bob, chanPointBobCarol, false,
+	)
+
+	// Get Bob's funding point.
+	bobChanTXID, err := lnrpc.GetChanPointFundingTxid(chanPointBobCarol)
+	require.NoError(t.t, err, "unable to get txid")
+	bobFundPoint := wire.OutPoint{
+		Hash:  *bobChanTXID,
+		Index: chanPointBobCarol.OutputIndex,
+	}
+
+	// We should have the following topology now,
+	// Alice <--public:100k--> Bob <--private:100k--> Carol
+	//
+	// Now we will create an invoice for Carol.
+	const paymentAmt = 20000
+	invoice := &lnrpc.Invoice{
+		Memo:    "routing hints",
+		Value:   paymentAmt,
+		Private: true,
+	}
+	ctxt, _ = context.WithTimeout(ctxb, defaultTimeout)
+	resp, err := carol.AddInvoice(ctxt, invoice)
+	require.NoError(t.t, err, "unable to create invoice for carol")
+
+	// Bob now updates the channel edge policy for the private channel.
+	const (
+		baseFeeMSat = 33000
+	)
+	timeLockDelta := uint32(chainreg.DefaultBitcoinTimeLockDelta)
+	updateFeeReq := &lnrpc.PolicyUpdateRequest{
+		BaseFeeMsat:   baseFeeMSat,
+		TimeLockDelta: timeLockDelta,
+		Scope: &lnrpc.PolicyUpdateRequest_ChanPoint{
+			ChanPoint: chanPointBobCarol,
+		},
+	}
+	ctxt, _ = context.WithTimeout(ctxb, defaultTimeout)
+	_, err = net.Bob.UpdateChannelPolicy(ctxt, updateFeeReq)
+	require.NoError(t.t, err, "unable to update chan policy")
+
+	// Alice pays the invoices. She will use the updated baseFeeMSat in the
+	// payment
+	ctxt, _ = context.WithTimeout(ctxb, defaultTimeout)
+	payReqs := []string{resp.PaymentRequest}
+	require.NoError(t.t,
+		completePaymentRequests(
+			ctxt, net.Alice, net.Alice.RouterClient, payReqs, true,
+		), "unable to send payment",
+	)
+
+	// Carol should have received 20k satoshis from Bob.
```

Comment:
We can add an assertion here that the payment was completed using two HTLCs, where the first failed with fee insufficient (you get the info from `ListPayments`)

Reply:
Good point, added the check to verify the status of the HTLCs.